src/SharpBrick.PoweredUp/Devices/RemoteControlButton.cs

```diff
@@ -0,0 +1,99 @@
+﻿using SharpBrick.PoweredUp.Protocol;
+using SharpBrick.PoweredUp.Utils;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+
+namespace SharpBrick.PoweredUp.Devices
+{
+    public class RemoteControlButton : Device, IPoweredUpDevice
+    {
+        protected SingleValueMode<sbyte> _remoteControlKeyMode;
+        protected SingleValueMode<sbyte> _keyAMode;
+        protected SingleValueMode<sbyte> _keyRMode;
+        protected SingleValueMode<sbyte> _keyDMode;
+        protected MultiValueMode<sbyte> _keySDMode;
+
+        public byte ModeIndexRemoteControlKey { get; protected set; } = 0;
+        public byte ModeIndexKeyA { get; protected set; } = 1;
+        public byte ModeIndexKeyR { get; protected set; } = 2;
+        public byte ModeIndexKeyD { get; protected set; } = 3;
+        public byte ModeIndexKeySD { get; protected set; } = 4;
+
+        public sbyte RemoteControlKey => _remoteControlKeyMode.SI;
+        public IObservable<sbyte> RemoteControlKeyObservable => _remoteControlKeyMode.Observable.Select(x => x.SI);
+
+        public sbyte KeyA => _keyAMode.SI;
+        public IObservable<sbyte> KeyAObservable => _keyAMode.Observable.Select(x => x.SI);
+
+        public sbyte KeyR => _keyRMode.SI;
+        public IObservable<sbyte> KeyRObservable => _keyRMode.Observable.Select(x => x.SI);
+
+        public sbyte KeyD => _keyDMode.SI;
+        public IObservable<sbyte> KeyDObservable => _keyDMode.Observable.Select(x => x.SI);
+
+        public sbyte[] KeySD => _keySDMode.SI;
```

Comment:
I do not like these properties. They do not tell anything. Especially considering that only KeySD is returning data for all of them (

Reply:
What I did in some other cases, I made the single enumerable split up into multiple streams and properties.

Reply:
But wait till we have digged a bit more about the other properties.

Reply:
Yeah, I can split this properly, with sensible names, there are three buttons:

- +
- red stop button
- -

I'm not sure about the common names used for these buttons though.
I'll wait with making any changes, don't have much time right now anyway :)

Reply:
- KEYSD seems to be a complete map (like you worked out
- RCKEY (Mode: 0) seems to be events (https://github.com/nathankellenicki/node-poweredup/blob/master/src/devices/remotecontrolbutton.ts)